modules/tensor_mechanics/doc/content/modules/tensor_mechanics/Slipweakening_draft.md

```diff
@@ -0,0 +1,1289 @@
+  const MaterialProperty<RankTwoTensor> & _stress;
+  Real _q;
+
+};
+```
+
+*Figure 4 StiffPropDamping: Header File*
+
+The source file implements the weak form evaluation at each quadrature point, here we follow similar definition of damping stress tensor given in (Day, 2005), Appendix A8:
+
+$$\begin{array}{r} \overline{\sigma} = \Delta t\left\lbrack \frac{\sigma_{t} - \sigma_{t - \Delta t}}{\Delta t} \right\rbrack = \left\lbrack \sigma_{t} - \sigma_{t - \Delta t} \right\rbrack (5)\end{array}$$
+
+Where $\sigma_{t}$ and $\sigma_{t - \Delta t}$ are stress tensor from current/last time step. The code snippet is given below, notice that for each custom function, user needs to register the function to their own app using ```registerMooseObject```. (```TensorMechanicsApp``` is a custom app name that can be replaced).
+
+```
+#include "StiffPropDamping.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("TensorMechanicsApp", StiffPropDamping);
+
+InputParameters
+StiffPropDamping::validParams()
+{
+  InputParameters params = StressDivergenceTensors::validParams();
+  params.addClassDescription(
+      "Compute Stiffness Proportional Damping Residual");
+
+  params.addParam<Real>("q", 0.1, "Ratio Factor to assign magnitude of stiffness proportional damping term");
+
+  return params;
+}
+
+StiffPropDamping::StiffPropDamping(const InputParameters & parameters)
+  : StressDivergenceTensors(parameters),
+  
+  //Get stress tensor from previous time step (older == previous time step in Explicit Time Integration Scheme)
+  _stress_older(getMaterialPropertyOlderByName<RankTwoTensor>(_base_name + "stress")),
+  
+  //Get stress tensor from current time step
+  _stress(getMaterialPropertyByName<RankTwoTensor>(_base_name + "stress")),
+  
+  //Ratio factor 
+  _q(getParam<Real>("q"))
+
+{
+}
+
+Real
+StiffPropDamping::computeQpResidual()
+{
+
+  Real residual = 0.0;
+
+  residual += _q*(_stress[_qp].row(_component)-_stress_older[_qp].row(_component)) * _grad_test[_i][_qp];
+
+  return residual;
+}
+```
+
+*Figure 5 StiffPropDamping: Source File*
+
+To utilize the kernel, allocate it inside ```[Kernels]``` section of input file:
+
+```
+[./Reactionx]
+    type = StiffPropDamping
+    variable = 'disp_x'
+    component = '0'
+[]
+[./Reactiony]
+    type = StiffPropDamping
+    variable = 'disp_y'
+    component = '1'
+[]
+[./Reactionz]
+    type = StiffPropDamping
+    variable = 'disp_z'
+    component = '2'
+[]
+```
+
+*Figure 6 StiffPropDamping: Input File*
+
+#### AuxKernels
+
+All the quantities passed as variable input into the material kernel are defined as aux-variables. The code for defining aux variables residuals (```resid, resid_slipweakening```), displacements (```disp_slipweakening```), velocity (```vel_slipweakening```), spatial distribution of static friction coefficient (```mu_s```) and initial shear stress along strike direction (```ini_shear_stress```) is given below:
+
+```
+[AuxVariables]
```

Comment:
instead of pasting input files in the md file, you can use this syntax:

!listing <path to input.i> block=AuxVariables

Reply:
I'm not familiar with this, like this?

!listing moose/modules/tensor_mechanics/test/tests/2D_slipweakening/tpv2052D.i 
block=Modules
id=input-block
caption=StressDivergenceTensors Kernels: Input File

Is there a way to preview it in the vscode? I may need to ask framework people about this